### DISCUSSION TITLE
[solved] [false alarm] docker PIDS keeps increasing at 7.0.8

### DISCUSSION THREAD

**Discussion by tamaszsellerp92** [docker-library/mongo#683] [General]
I'm running a plain and simple 7.0.8 image with a modified config

storage:
    dbPath: /data/db
    directoryPerDB: true
    engine: wiredTiger
    wiredTiger:
        engineConfig:
            directoryForIndexes: true

and the PIDS number in docker stats keeps increasing rapidly: 14696 for a 3 day old container
```
CONTAINER ID   NAME             CPU %     MEM USAGE / LIMIT     MEM %     NET I/O          BLOCK I/O         PIDS
230acbbcf833   serene_mestorf   101.39%   7.859GiB / 23.47GiB   33.49%    50.7GB / 481GB   6.46GB / 17.4GB   14696

```
the ps within the docker container shows only 3 processes, their pids are constant

what is going on? only the mongodb container is raising the PIDS number.
Healthcheck is not enabled on this container.

**Comment by tianon** [comment]
That is definitely an extremely large number -- I don't think it's even cumulative. :grimacing:

The difference between `ps`/`top` and that number is probably threads -- if you use something like `docker top mongo -eLf` you'll probably (hopefully) see a much longer list.

Is this instance exposed to the internet?  I think every incoming connection generates a thread, and that could account for such a huge number (but would also be terrifying :grimacing:).

**Reply by tamaszsellerp92** [reply]
thanks for your reply!
Indeed, docker top .. -eLf gave me the complete list, which is full of :
```
config /configs/mongodb.conf --bind_ip_all
lxd                 994437              994411              1015421             0                   167                 04:56               ?                   00:00:00            mongod --config /configs/mongodb.conf --bind_ip_all

```

This instance is not exposed to the internet, it's on a private network but there is an application connected to it, which may open regular connections to it.

**Comment by tamaszsellerp92** [comment]
okay, I think I got it!
 probably it has nothing to do with mongo, but with the new function of our application.
the developer does not close the mongodb connection and it's kept open, that's why the threads are increasing!

the number of open connections keeps increasing... LOL.

false alarm, sorry guys.
